autofit/non_linear/grid/sensitivity.py

```diff
@@ -4,100 +4,88 @@
+        self.perturb_result = perturb_result
 
     @property
     def log_likelihood_difference(self):
-        return self.perturbed_result.log_likelihood - self.result.log_likelihood
+        return self.perturb_result.log_likelihood - self.result.log_likelihood
 
     @property
     def log_likelihood_base(self):
         return self.result.log_likelihood
 
     @property
     def log_likelihood_perturbed(self):
-        return self.perturbed_result.log_likelihood
+        return self.perturb_result.log_likelihood
 
 
 class Job(AbstractJob):
     _number = count()
 
-    use_instance = False
-
     def __init__(
         self,
-        analysis_factory: "AnalysisFactory",
         model: AbstractPriorModel,
-        perturbation_model: AbstractPriorModel,
+        simulate_cls: Callable,
+        perturb_model: AbstractPriorModel,
+        simulate_instance: ModelInstance,
         base_instance: ModelInstance,
-        perturbation_instance: ModelInstance,
-        search: NonLinearSearch,
+        base_fit_cls: Callable,
+        perturb_fit_cls: Callable,
+        paths: AbstractPaths,
         number: int,
     ):
         """
-        Job to run non-linear searches comparing how well a model and a model with a perturbation
-        fit the image.
+        Job to run non-linear searches comparing how well a model and a model with a perturbation fit the image.
 
         Parameters
         ----------
         model
             A base model that fits the image without a perturbation
-        perturbation_model
+        perturb_model
             A model of the perturbation which has been added to the underlying image
-        analysis_factory
-            Factory to generate analysis classes which comprise a model and data
-        search
-            A non-linear search
+        base_fit_cls
+            A class which defines the function which fits the base model to each simulated dataset of the sensitivity
+            map.
+        perturb_fit_cls
+            A class which defines the function which fits the perturbed model to each simulated dataset of the
+            sensitivity map.
+        paths
+            The paths defining the output directory structure of the sensitivity mapping.
         """
         super().__init__(number=number)
 
-        self.analysis_factory = analysis_factory
         self.model = model
-
-        self.perturbation_model = perturbation_model
+        self.simulate_cls = simulate_cls
+        self.perturb_model = perturb_model
+        self.simulate_instance = simulate_instance
         self.base_instance = base_instance
-        self.perturbation_instance = perturbation_instance
-
-        self.search = search.copy_with_paths(
-            search.paths.for_sub_analysis(
-                "[base]",
-            )
-        )
-        self.perturbed_search = search.copy_with_paths(
-            search.paths.for_sub_analysis(
-                "[perturbed]",
-            )
-        )

```

Comment:
This is one part of your problem. I guess you need to manually update this in your 'Fit' classes